ddtrace/internal/logger.py

```diff
@@ -125,6 +125,13 @@ def handle(self, record):
         :param record: The log record being logged
         :type record: ``logging.LogRecord``
         """
+        if record.levelno >= logging.ERROR:
+            # avoid circular import
+            from ddtrace.internal.telemetry import telemetry_writer
```

Comment:
why doesn't telemetry writer just add a log handler on `ddtrace` logger?

Reply:
That's a wayyyyy better solution. Thanks for the tip

Reply:
Logging handlers are mainly used to write logs to different sources. By default handlers acquire an [i/o lock](https://github.com/python/cpython/blob/3.11/Lib/logging/__init__.py#L965) before handling a log. This is expensive and is not necessary to intercept and store log messages for the telemetry writer. Instead of using a stream or file handler I could implement the [nullhandler](https://docs.python.org/3/library/logging.handlers.html#nullhandler) but this is not supported in python 2.7. I could override some of this logic in the file/stream handlers to improve performance but I don't think it's worth the effort.

Another approach could be to define a telemetry log filter. One limitation of this approach is that log filters are not automatically applied to descendant loggers. Ensuring the filter is configured correctly could be a pain.

> Note that filters attached to handlers are consulted before an event is emitted by the handler, whereas filters attached to loggers are consulted whenever an event is logged (using [debug()](https://docs.python.org/3/library/logging.html#logging.debug), [info()](https://docs.python.org/3/library/logging.html#logging.info), etc.), before sending an event to handlers. This means that events which have been generated by descendant loggers will not be filtered by a logger’s filter setting, unless the filter has also been applied to those descendant loggers. ([link](https://docs.python.org/3/library/logging.html#logging.Filter.filter))



Hmmm I can't think of a better alternative. I think this approach mixes concerns but I think it's the easiest to read, maintain, and gives us the best performance. Thoughts? @brettlangdon